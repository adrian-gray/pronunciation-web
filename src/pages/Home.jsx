import React from "react";

import Paper from "@material-ui/core/Paper";
import Typography from "@material-ui/core/Typography";
import { makeStyles } from "@material-ui/core/styles";
import { withTheme } from "@material-ui/styles";

import data from "./../data/sounds";
import SEO from "./../components/SEO";
import Phoneme from "./../components/Phoneme";

const useStyles = makeStyles(theme => ({
  page: theme.page
}));

const Sounds = props => {
  const classes = useStyles(props);

  const phonemes = data.phonemes;
  const list = Object.keys(phonemes);
  const cards = list.map(phoneme => {
    const ipa = phonemes[phoneme]["ipa"];
    const title = phonemes[phoneme]["title"];
    const tag = phonemes[phoneme]["tag"];
    const words = phonemes[phoneme]["activities"]["common words"].words;
    return (
      <Phoneme
        key={phoneme}
        phoneme={phoneme}
        ipa={ipa}
        title={title}
        tag={tag}
        words={words}
      />
    );
  });

  return (
    <Paper className={classes.page}>
      <SEO meta="sounds" />
      <Typography variant="h4" gutterBottom>
        {"Phoenetic English Pronunciation"}
      </Typography>
      {cards}
    </Paper>
  );
};

export default withTheme(Sounds);
