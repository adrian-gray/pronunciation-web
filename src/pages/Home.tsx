import React from "react";

import Container from "react-bootstrap/Container";

import soundsData from "./../data/sounds.js";
import SEO from "../components/SEO";
import Phoneme from "../components/Phoneme";

export default () => {
  const phonemes = soundsData;
  const list = Object.keys(phonemes);
  const cards = list.map(phoneme => {
    const ipa = phonemes[phoneme]["ipa"];
    const title = phonemes[phoneme]["title"];
    const tag = phonemes[phoneme]["tag"];
    const words = phonemes[phoneme]["activities"]["common words"].words;
    return (
      <Phoneme
        key={phoneme}
        phoneme={phoneme}
        ipa={ipa}
        title={title}
        tag={tag}
        words={words}
      />
    );
  });

  return (
    <Container className="page">
      <SEO meta="sounds" />
      <h1>{"Phoenetic English Pronunciation"}</h1>
      {cards}
    </Container>
  );
};
