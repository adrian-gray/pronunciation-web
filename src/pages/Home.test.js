import React from "react";
import { shallow } from "enzyme";
import { render, cleanup } from "@testing-library/react";

import Home from "./Home";

afterEach(cleanup);

test("Home", async () => {
  const props = {};

  const container = shallow(<Home {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
