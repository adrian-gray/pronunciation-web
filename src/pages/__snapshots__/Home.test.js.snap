// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home 1`] = `
<div>
  <div
    class="page"
  >
    <h1>
      Phoenetic English Pronunciation
    </h1>
    <div
      class="headspace"
    >
      <div
        class="mini-canvas"
      >
        <h3>
          <span>
            <span />
            <span
              class="hilite"
            >
              /u:/
            </span>
            <span />
             
             – 
            <span>
              long 
            </span>
            <span
              class="hilite"
            >
              u
            </span>
            <span>
               ‘bl
            </span>
            <span
              class="hilite"
            >
              ue
            </span>
            <span>
              ’
            </span>
             
          </span>
        </h3>
        <p>
          Example words containing short ‘a’
        </p>
        <p>
          <span
            class="phoneme-word"
          >
            <span>
              st
            </span>
            <span
              class="hilite"
            >
              u
            </span>
            <span>
              dent
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              J
            </span>
            <span
              class="hilite"
            >
              u
            </span>
            <span>
              ne
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              j
            </span>
            <span
              class="hilite"
            >
              ew
            </span>
            <span>
              ellery (jewelry)
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              m
            </span>
            <span
              class="hilite"
            >
              u
            </span>
            <span>
              sic
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              n
            </span>
            <span
              class="hilite"
            >
              ew
            </span>
            <span />
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              f
            </span>
            <span
              class="hilite"
            >
              ew
            </span>
            <span />
             
          </span>
        </p>
        <a
          class="undecorated"
          href="/sound/long-u"
        >
          <h5>
            Click to learn more about the long ‘u’ sound
          </h5>
        </a>
      </div>
    </div>
    <div
      class="headspace"
    >
      <div
        class="mini-canvas"
      >
        <h3>
          <span>
            <span />
            <span
              class="hilite"
            >
              /æ/
            </span>
            <span />
             
             – 
            <span>
              short 
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
               ‘b
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              d’
            </span>
             
          </span>
        </h3>
        <p>
          Example words containing short ‘a’
        </p>
        <p>
          <span
            class="phoneme-word"
          >
            <span>
              c
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              sh
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              tr
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              sh
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              tr
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              nsfer
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              st
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              nd
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              ch
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              t
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              tr
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              ffic
            </span>
             
          </span>
        </p>
        <a
          class="undecorated"
          href="/sound/short-a"
        >
          <h5>
            Click to learn more about the short ‘a’ sound
          </h5>
        </a>
      </div>
    </div>
    <div
      class="headspace"
    >
      <div
        class="mini-canvas"
      >
        <h3>
          <span>
            <span />
            <span
              class="hilite"
            >
              /eɪ/
            </span>
            <span />
             
             – 
            <span>
              long 
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
               ‘n
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              me’
            </span>
             
          </span>
        </h3>
        <p>
          Example words containing short ‘a’
        </p>
        <p>
          <span
            class="phoneme-word"
          >
            <span>
              m
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              ke
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              ch
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              nge
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              p
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              per
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              w
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              y
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              f
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              ce
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              g
            </span>
            <span
              class="hilite"
            >
              a
            </span>
            <span>
              me
            </span>
             
          </span>
        </p>
        <a
          class="undecorated"
          href="/sound/long-a"
        >
          <h5>
            Click to learn more about the long ‘a’ sound
          </h5>
        </a>
      </div>
    </div>
    <div
      class="headspace"
    >
      <div
        class="mini-canvas"
      >
        <h3>
          <span>
            <span />
            <span
              class="hilite"
            >
              /i:/
            </span>
            <span />
             
             – 
            <span>
              long 
            </span>
            <span
              class="hilite"
            >
              e
            </span>
            <span>
               ‘s
            </span>
            <span
              class="hilite"
            >
              ee
            </span>
            <span>
              ’
            </span>
             
          </span>
        </h3>
        <p>
          Example words containing short ‘a’
        </p>
        <p>
          <span
            class="phoneme-word"
          >
            <span>
              pl
            </span>
            <span
              class="hilite"
            >
              ea
            </span>
            <span>
              se
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              k
            </span>
            <span
              class="hilite"
            >
              ey
            </span>
            <span />
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              tr
            </span>
            <span
              class="hilite"
            >
              ee
            </span>
            <span />
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              n
            </span>
            <span
              class="hilite"
            >
              ee
            </span>
            <span>
              d
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              sl
            </span>
            <span
              class="hilite"
            >
              ee
            </span>
            <span>
              p
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span />
            <span
              class="hilite"
            >
              ea
            </span>
            <span>
              st
            </span>
             
          </span>
        </p>
        <a
          class="undecorated"
          href="/sound/long-e"
        >
          <h5>
            Click to learn more about the long ‘e’ sound
          </h5>
        </a>
      </div>
    </div>
    <div
      class="headspace"
    >
      <div
        class="mini-canvas"
      >
        <h3>
          <span>
            <span />
            <span
              class="hilite"
            >
              /ɪ/
            </span>
            <span />
             
             – 
            <span>
              short 
            </span>
            <span
              class="hilite"
            >
              i
            </span>
            <span>
               ‘s
            </span>
            <span
              class="hilite"
            >
              i
            </span>
            <span>
              t’
            </span>
             
          </span>
        </h3>
        <p>
          Example words containing short ‘a’
        </p>
        <p>
          <span
            class="phoneme-word"
          >
            <span>
              th
            </span>
            <span
              class="hilite"
            >
              i
            </span>
            <span>
              nk
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span />
            <span
              class="hilite"
            >
              i
            </span>
            <span>
              nternet
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              m
            </span>
            <span
              class="hilite"
            >
              i
            </span>
            <span>
              nutes
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              g
            </span>
            <span
              class="hilite"
            >
              i
            </span>
            <span>
              ve
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              qu
            </span>
            <span
              class="hilite"
            >
              i
            </span>
            <span>
              ck
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              br
            </span>
            <span
              class="hilite"
            >
              i
            </span>
            <span>
              dge
            </span>
             
          </span>
        </p>
        <a
          class="undecorated"
          href="/sound/short-i"
        >
          <h5>
            Click to learn more about the short ‘i’ sound
          </h5>
        </a>
      </div>
    </div>
    <div
      class="headspace"
    >
      <div
        class="mini-canvas"
      >
        <h3>
          <span>
            <span />
            <span
              class="hilite"
            >
              /Λ/
            </span>
            <span />
             
             – 
            <span>
              short 
            </span>
            <span
              class="hilite"
            >
              u
            </span>
            <span>
               ‘bl
            </span>
            <span
              class="hilite"
            >
              ue
            </span>
            <span>
              ’
            </span>
             
          </span>
        </h3>
        <p>
          Example words containing short ‘a’
        </p>
        <p>
          <span
            class="phoneme-word"
          >
            <span>
              j
            </span>
            <span
              class="hilite"
            >
              u
            </span>
            <span>
              st
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              en
            </span>
            <span
              class="hilite"
            >
              ou
            </span>
            <span>
              gh
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              c
            </span>
            <span
              class="hilite"
            >
              o
            </span>
            <span>
              lour (color)
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              n
            </span>
            <span
              class="hilite"
            >
              u
            </span>
            <span>
              mber
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              l
            </span>
            <span
              class="hilite"
            >
              o
            </span>
            <span>
              ve
            </span>
             
          </span>
          <span
            class="phoneme-word"
          >
            <span>
              tr
            </span>
            <span
              class="hilite"
            >
              ou
            </span>
            <span>
              ble
            </span>
             
          </span>
        </p>
        <a
          class="undecorated"
          href="/sound/short-u"
        >
          <h5>
            Click to learn more about the short ‘u’ sound
          </h5>
        </a>
      </div>
    </div>
  </div>
</div>
`;
